WordPress/WordPressTest/WPCrashLoggingDataProviderTests.swift

```diff
@@ -0,0 +1,79 @@
+import XCTest
+import AutomatticTracks
+
+@testable import WordPress
+
+final class WPCrashLoggingDataProviderTests: XCTestCase {
+
+    // MARK: - Testing Log Error
+
+    func testReadingTrackUserInMainThread() throws {
+        // Given
+        let dataProvider = self.makeCrashLoggingDataProvider()
+
+        // When
+        let user = try XCTUnwrap(dataProvider.currentUser)
+
+        // Then
+        XCTAssertEqual(user.userID, "\(Constants.defaultAccountID)")
+        XCTAssertEqual(user.username, Constants.defaultAccountUsername)
+        XCTAssertEqual(user.email, Constants.defaultAccountEmail)
+    }
+
+    func testReadingTrackUserInBackgroundThread() {
+        // Given
+        let expectation = XCTestExpectation(description: "Should return current user")
+        let dataProvider = self.makeCrashLoggingDataProvider()
+        let dispatchGroup = DispatchGroup()
+        let numberOfOperations = 1000
+
+        // When
+        for _ in 0..<numberOfOperations {
+            dispatchGroup.enter()
+            DispatchQueue.global().async {
+                defer {
+                    dispatchGroup.leave()
+                }
+                guard let user = dataProvider.currentUser else {
+                    XCTFail("Failed to unwrap `dataProvider.currentUser`")
+                    return
+                }
+                XCTAssertEqual(user.userID, "\(Constants.defaultAccountID)")
+                XCTAssertEqual(user.username, Constants.defaultAccountUsername)
+                XCTAssertEqual(user.email, Constants.defaultAccountEmail)
+            }
+        }
+
+        // Then
+        dispatchGroup.notify(queue: .main) {
+            expectation.fulfill()
+        }
+        wait(for: [expectation], timeout: 2)
+    }
```

Comment:
Instead of testing `WPCrashLoggingDataProvider.currentUser` thread-safe access, I wanted to test instead `CrashLogging.logError` thread-safety, but that method doesn't return anything to assert the correctness of the event sent. It would require making changes to `CrashLogging` implementation in `Automattic-Tracks-iOS` repo, which is not the scope of this PR.

Reply:
Opened https://github.com/Automattic/Automattic-Tracks-iOS/issues/259 to track this